EIPS/eip-ondemand-data.md

```diff
@@ -57,6 +66,8 @@ Another major advantage of this standard is that oracles can charge fees in the
 
 To prevent data becoming too stale for a request between the simulation and a call's execution, ideally a contract could also emit the following event: `event OracleDataUsed(bytes32 oracleId, bytes oracleQuery, address oracleContract, uint expirationTime)` Here, `expirationTime` is the time after which the `OracleDataRequired` error would be thrown by the contract. (This would typically be a calculation involving a staleness tolerance and `block.timestamp`). Client applications that implement this standard would be able to recognize this event during simulation and estimate if an additional update will still be necessary, taking into account the speed of the chain. For example, the oracle query may request the latest quote available for a particular price feed and the expiration time may signal that the price cannot be older than three seconds prior to the current timestamp recognized by the blockchain. This has been omitted from the standard because there isn't a practical way to retrieve event data during transaction simulations on most JSON-RPC APIs at this time.
 
+It is the responsibility of the client to decide how to construct the multicall. Wallets that support account abstraction (per [ERC-4337](./eip-4337.md)) should already have the ability to generate atomic multi-operations. For EOA support, a multicall contract may be used. This may need to be a protocol-specific multicall proxy contract if the function being called references `msg.sender`. Here, the protocol should honor the reported `msg.sender` value passed from the multicall proxy contract.
```

Comment:
how do you think clients will implement this in practice? it seems like they will now need to know how each protocol supports multicalls, which will require another registry similar to the oracle one. 

if there's a way for the contract to suggest to the client how to do the multicall, that could help make this more generic.

Reply:
Yea, we discussed potentially having a multicall address in the revert. Realistically, I think support for this will all be built in at the application layer until wallets support account abstraction, but it's hard to say.

Reply:
yeah... i guess if you think we're going to get account abstraction soon, then it's a moot point.

Reply:
Another solution could be valuable, but I think it'd be a different EIP? Like if the `CrosschainAdder` contract had functions with `msg.sender`, it would need to be able to communicate 'use `x` function to multicall me' (and it may want to do this for reasons other than this standard too). The `OracleContract` implementing this standard wouldn't be able to help. (Or maybe there's something clever I'm not considering...)

Reply:
Practically, I was imagining there'd be a community-maintained TS library that integrates with viem/ethers.js/etc. There'd be an extendable config for oracle id resolution and an extendable config with a mapping of chainId:contractAddress => multicall func. If you're trying to call a contract that isn't in mapping, you'd need to specify that it should use the generic multicall contract or a custom one.